### DISCUSSION TITLE
Alternative naming strategy for default builder methods

### DISCUSSION THREAD

**Discussion by andrej-urvantsev** [Randgalt/record-builder#192] [Ideas]
Right now, if we have for example record named `DeliveryEvent`, then generated builder will have name `DeliveryEventBuilder` with static builder method `builder()`, so it will be used in the code like this:

```java
DeliveryEventBuilder.builder().foo(xx).bar(yy).build()
```

which looks a bit ambiguous.

What if we instead could set alternative default(meaning that it should be possible to set it in `RecordBuilder.Template`) naming strategy which would generate builder methods which correspond to builder class name, then it will be possible to use static imports and code above instead will be written as:

```java
deliveryEventBuilder().foo(xx).bar(yy).build()
```

**Comment by Randgalt** [comment]
We already have an option to change the name of the builder. I guess it could optionally have some kind of substitution mechanism. This is the first request for it, though. Doesn't seem horizontally useful.
